L1Trigger/TrackFindingTracklet/src/TrackDerTable.cc

```diff
@@ -0,0 +1,1478 @@
+                       (1 << settings->fitphi0bitshift());
+    iD[2][2 * i + 1] =
+        D[2][2 * i + 1] * (1 << settings->chisqzfactbits()) * settings->ktpars() / (1 << settings->fittbitshift());
+    iD[3][2 * i + 1] =
+        D[3][2 * i + 1] * (1 << settings->chisqzfactbits()) * settings->kzpars() / (1 << settings->fitz0bitshift());
+
+    //First the barrel
+    if (i < nlayers) {
+      MinvDt[0][2 * i] *= rnew[i] / sigmax;
+      MinvDt[1][2 * i] *= rnew[i] / sigmax;
+      MinvDt[2][2 * i] *= rnew[i] / sigmax;
+      MinvDt[3][2 * i] *= rnew[i] / sigmax;
+
+      iMinvDt[0][2 * i] =
+          (1 << settings->fitrinvbitshift()) * MinvDt[0][2 * i] * settings->kphi1() / settings->krinvpars();
+      iMinvDt[1][2 * i] =
+          (1 << settings->fitphi0bitshift()) * MinvDt[1][2 * i] * settings->kphi1() / settings->kphi0pars();
+      iMinvDt[2][2 * i] = (1 << settings->fittbitshift()) * MinvDt[2][2 * i] * settings->kphi1() / settings->ktpars();
+      iMinvDt[3][2 * i] = (1 << settings->fitz0bitshift()) * MinvDt[3][2 * i] * settings->kphi1() / settings->kzpars();
+
+      if (rnew[i] < 60.0) {
+        MinvDt[0][2 * i + 1] /= sigmazpsbarrel;
+        MinvDt[1][2 * i + 1] /= sigmazpsbarrel;
+        MinvDt[2][2 * i + 1] /= sigmazpsbarrel;
+        MinvDt[3][2 * i + 1] /= sigmazpsbarrel;
+
+        iMinvDt[0][2 * i + 1] =
+            (1 << settings->fitrinvbitshift()) * MinvDt[0][2 * i + 1] * settings->kzproj() / settings->krinvpars();
+        iMinvDt[1][2 * i + 1] =
+            (1 << settings->fitphi0bitshift()) * MinvDt[1][2 * i + 1] * settings->kzproj() / settings->kphi0pars();
+        iMinvDt[2][2 * i + 1] =
+            (1 << settings->fittbitshift()) * MinvDt[2][2 * i + 1] * settings->kzproj() / settings->ktpars();
+        iMinvDt[3][2 * i + 1] =
+            (1 << settings->fitz0bitshift()) * MinvDt[3][2 * i + 1] * settings->kzproj() / settings->kzpars();
+      } else {
+        MinvDt[0][2 * i + 1] /= sigmaz2;
+        MinvDt[1][2 * i + 1] /= sigmaz2;
+        MinvDt[2][2 * i + 1] /= sigmaz2;
+        MinvDt[3][2 * i + 1] /= sigmaz2;
+
+        int fact = (1 << (settings->nzbitsstub(0) - settings->nzbitsstub(5)));
+
+        iMinvDt[0][2 * i + 1] = (1 << settings->fitrinvbitshift()) * MinvDt[0][2 * i + 1] * fact * settings->kzproj() /
+                                settings->krinvpars();
+        iMinvDt[1][2 * i + 1] = (1 << settings->fitphi0bitshift()) * MinvDt[1][2 * i + 1] * fact * settings->kzproj() /
+                                settings->kphi0pars();
+        iMinvDt[2][2 * i + 1] =
+            (1 << settings->fittbitshift()) * MinvDt[2][2 * i + 1] * fact * settings->kzproj() / settings->ktpars();
+        iMinvDt[3][2 * i + 1] =
+            (1 << settings->fitz0bitshift()) * MinvDt[3][2 * i + 1] * fact * settings->kzproj() / settings->kzpars();
+      }
+    }
+
+    //Secondly the disks
+    else {
+      if (std::abs(alpha[i - nlayers]) < 1e-10) {
```

Comment:
simplify: `denom = (std::abs(alpha[i - nlayers]) < 1e-10) ? sigmaxpsdisk : sigmax2sdisk;`, then divide by denom
(or even make denomInv and multiply by it)

Reply:
implemented